feature/interface/holdtime/otg_tests/holdtime_test/holddown_timers_test.go

```diff
@@ -0,0 +1,670 @@
+	dstEth.Connection().SetLagName(agg.Name())
+	dstEth.Ipv4Addresses().Add().SetName(lagName + ".IPv4").SetAddress(ateDst.IPv4).SetGateway(dutDst.IPv4).SetPrefix(uint32(ateDst.IPv4Len))
+}
+
+func displaySummaryTable(t *testing.T,
+	preActionTS,
+	postActionTS string,
+	actualDurationInMS,
+	minToleranceInMS,
+	maxToleranceInMS int64,
+	expectedOperStatus,
+	actualOperStatus string,
+	pass bool) {
+
+	result := "FAIL"
+	if pass {
+		result = "PASS"
+	}
+
+	// Prepare the strings for output.
+	expectedDurationStr := "300ms" // Assuming this is a constant value
+	minToleranceStr := fmt.Sprintf("%dms", minToleranceInMS)
+	maxToleranceStr := fmt.Sprintf("%dms", maxToleranceInMS)
+	actualDurationStr := fmt.Sprintf("%dms", actualDurationInMS)
+
+	// Create a slice of metrics and corresponding values.
+	metrics := []string{"Pre-action TS", "Post-action TS", "Expected Duration", "Actual Duration", "Min Tolerance", "Max Tolerance", "Expected Oper Status", "Actual Oper Status", "Result"}
+	values := []string{preActionTS, postActionTS, expectedDurationStr, actualDurationStr, minToleranceStr, maxToleranceStr, expectedOperStatus, actualOperStatus, result}
+
+	// Find the maximum width for the metrics to align the values.
+	maxMetricWidth := 0
+	for _, metric := range metrics {
+		if len(metric) > maxMetricWidth {
+			maxMetricWidth = len(metric)
+		}
+	}
+
+	// Create the vertical table.
+	table := ""
+	for i, metric := range metrics {
+		table += fmt.Sprintf("%-*s: %s\n", maxMetricWidth, metric, values[i])
+	}
+
+	t.Logf("\n%s", table)
+}
+
+func flapOTGInterface(t *testing.T,
+	ate *ondatra.ATEDevice,
+	dut *ondatra.DUTDevice,
+	actionState string) (time.Time, time.Time, string, string) {
+
+	// Shut down OTG Interface
+	p1 := ondatra.ATE(t, "ate").Port(t, "port1")
+	portStateAction := gosnappi.NewControlState()
+
+	var otgStateChangeTsStr string
+
+	// TC2 Step 1 Read timestamp of last oper-status change  form DUT port-1
+	preStateTSSTR := gnmi.Get(t, dut, gnmi.OC().Interface(aggID).LastChange().State())
+	DutLastChangeTS1 := time.Unix(0, int64(preStateTSSTR)).UTC().Format(time.RFC3339Nano)
+	t.Logf("Step1. DutLastChangeTS1 is: %v", DutLastChangeTS1)
+	if actionState == "UP" {
+		portStateAction.Port().Link().SetPortNames([]string{p1.ID()}).SetState(gosnappi.StatePortLinkState.UP)
+		otgStateChangeTsStr = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+		ate.OTG().SetControlState(t, portStateAction)
+	} else if actionState == "DOWN" {
+		// TC2 Step 2 Bring Down OTG Interface
+		t.Log("Step 2 sending remote fault")
+		portStateAction.Port().Link().SetPortNames([]string{p1.ID()}).SetState(gosnappi.StatePortLinkState.DOWN)
+		//otgStateChangeTsStr = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+		otgStateChangeTsStr = time.Now().UTC().Format(time.RFC3339Nano)
+
+		ate.OTG().SetControlState(t, portStateAction)
+
+		// TC2 Step 3
+		t.Log("Step 3 sleeping 1000ms")
+		time.Sleep(1000 * time.Millisecond)
```

Comment:
Sleep timers can introduce test flakiness.   What is the purpose of this sleep?  Can we change this to a event based trigger? For example, block on gnmi.Await for telemetry from the DUT port state change?